service/bfg/bfg.go

```diff
@@ -1006,8 +1012,9 @@ func (s *Server) deleteSession(id string) {
 }
 
 func (s *Server) handleWebsocketPrivate(w http.ResponseWriter, r *http.Request) {
-	log.Tracef("handleWebsocketPrivate: %v", r.RemoteAddr)
-	defer log.Tracef("handleWebsocketPrivate exit: %v", r.RemoteAddr)
+	remoteAddr := s.remoteIP(r)
```

Comment:
are we handling for a case where `remoteAddr == ""` in any uses of `s.remoteIP(...)`?

Reply:
If `remoteIP` returns `""` then `req.RemoteAddr` is invalid (either not a hostport, or not a valid IP address).
I don't think that can ever happen, and if it does then something is very wrong.

Reply:
> and if it does then something is very wrong.

should we error in this case?